smartlearner/views.py

```diff
@@ -51,3 +51,39 @@ def mean(self):
     @property
     def stderror(self):
         return ItemGetter(self, attribute=1)
+
+
+class ReconstructionError(View):
+    def __init__(self, predict_fct, dataset):
+        super().__init__()
+
+        batch_size = 1024  # Internal buffer
+        self.nb_batches = int(np.ceil(len(dataset) / batch_size))
+
+        input = dataset.symb_inputs
+        target = dataset.symb_targets
+        reconstruction_errors = T.sqr(predict_fct(input) - target).mean()
```

Comment:
Ah, you take the mean here. I think you want to take the mean along axis=1 and not across the examples of the batch (this one will be done below).